src/index.mjs

```diff
@@ -0,0 +1,209 @@
+import * as zx from "zx";
+import { strictEqual as assertEqual, ok as assertOk } from "assert";
+import fs from "fs";
+import env from "env-var";
+import "dotenv/config";
+
+const GEOSERVER_USER = env.get("GEOSERVER_USER").default("admin").asString();
+const GEOSERVER_PASS = env
+  .get("GEOSERVER_PASS")
+  .default("geoserver")
+  .asString();
+
+const DATA_STORE_CREDENTIALS = {
+  host: env.get("DATA_STORE_HOST").default("localhost").asString(),
+  port: env.get("DATA_STORE_PORT").default("5432").asString(),
+  database: env.get("DATA_STORE_DATABASE").default("postgres").asString(),
+  user: env.get("DATA_STORE_USER").default("postgres").asString(),
+  password: env.get("DATA_STORE_PASSWORD").default("postgres").asString(),
+  dbtype: env.get("DATA_STORE_DBTYPE").default("postgis").asString(),
+  schema: env.get("DATA_STORE_SCHEMA").default("polygon_parts").asString(),
+  ssl: env.get("DATA_STORE_SSL").default("DISABLE").asString(),
+};
+
+const GEOSERVER_BASE_URL = env
+  .get("GEOSERVER_BASE_URL")
+  .default("http://localhost:8080/geoserver")
+  .asString();
+
+const POLYGON_PARTS_WORKSPACE_NAME = env
+  .get("POLYGON_PARTS_WORKSPACE_NAME")
+  .default("polygon_parts")
+  .asString();
+
+const WORKSPACE_API_URL = `${GEOSERVER_BASE_URL}/rest/workspaces`;
+const DATA_STORE_API_URL = `${WORKSPACE_API_URL}/${POLYGON_PARTS_WORKSPACE_NAME}/datastores`;
+const DATA_STORE_NAME = env
+  .get("DATA_STORE_NAME")
+  .default("polygon_parts")
+  .asString();
+
+const FEATURE_TYPES_API_URL = `${DATA_STORE_API_URL}/${DATA_STORE_NAME}/featuretypes`;
+const LAYER_BODY_JSON = "./artifacts/polygonPartSpec.json";
+const MAX_FEATURES = env.get("MAX_FEATURES").default("0").asInt();
+const NUM_DECIMALS = env.get("NUM_DECIMALS").default("0").asInt();
+const LAYER_TITLE_NAME = env
+  .get("LAYER_TITLE_NAME")
+  .default("polygon_parts")
+  .asString();
+
+// will check if geoserver is up before running configuration
+while (true) {
```

Comment:
can you please create a function for each step with meeningful name? instead of putting all the code in a flat file?